File: Python/A_Project/B_Snake_and_Food/Utility/game_board.py
class GameBoard:
    _instance = None

    def __init__(self, width: int, height: int):
        if width <= 0 or height <= 0:
            raise ValueError("width and height must be positive")
        self.width = width
        self.height = height

    @classmethod
    def get_instance(cls, width: int, height: int):
        if cls._instance is None:
            cls._instance = cls(width, height)
        elif cls._instance.width != width or cls._instance.height != height:
            raise ValueError(
                f"GameBoard already initialized as "
                f"{cls._instance.width}x{cls._instance.height}, got {width}x{height}"
            )
        return cls._instance

    def get_width(self) -> int:
        return self.width

    def get_height(self) -> int:
        return self.height

File: Ruby/A_Project/B_Snake_And_Food/Utility/game_board.rb
# frozen_string_literal: true

class GameBoard
  class << self
    def get_instance(width, height)
      if @instance.nil?
        @instance = new(width, height)
      elsif @instance.width != width || @instance.height != height
        raise ArgumentError, "GameBoard already initialized as #{@instance.width}x#{@instance.height}, got #{width}x#{height}"
      end

      @instance
    end
  end

  attr_reader :width, :height

  def initialize(width, height)
    raise ArgumentError, "width and height must be positive" if width <= 0 || height <= 0

    @width = width
    @height = height
  end

  def get_width
    @width
  end

  def get_height
    @height
  end
end
